bugsnag/src/main/java/com/bugsnag/callbacks/ServletCallback.java

```diff
@@ -61,10 +62,20 @@ private String getClientIp(HttpServletRequest request) {
 
     private Map<String, String> getHeaderMap(HttpServletRequest request) {
         Map<String, String> map = new HashMap<String, String>();
-        Enumeration headerNames = request.getHeaderNames();
+        Enumeration<String> headerNames = request.getHeaderNames();
         while (headerNames.hasMoreElements()) {
-            String key = (String) headerNames.nextElement();
-            map.put(key, request.getHeader(key));
+            String key = headerNames.nextElement();
+            Enumeration<String> headerValues = request.getHeaders(key);
+            StringBuilder value = new StringBuilder(headerValues.nextElement());
+
+            // If there are multiple values for the header, do comma-separated concat
+            // as per RFC 2616:
+            // https://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.2
+            while (headerValues.hasMoreElements()) {
+                value.append(",").append(headerValues.nextElement());
+            }
+
+            map.put(key, value.toString());
```

Comment:
We should probably replace the value of certain headers with a placeholder (e.g. Authorisation, other sensitive keys).

Reply:
I've added Authorization to the filters as this is a registered header (see https://www.iana.org/assignments/message-headers/message-headers.xhtml). I'm not going to add any more at this stage as I think this needs wider discussion.